### DISCUSSION TITLE
is there any way to have loops?

### DISCUSSION THREAD

**Discussion by helalaou** [dolanmiu/docx#1587] [Q&A] [answered]
I tried everything that came to my mind (string manipulation using loops/ recursion and sending one var at a time) but it doesn't work because sending an instance of a docx as a function argument. doesn't work either.
I hope you can help me with this.  Thanks!

**Comment by dolanmiu** [comment, marked as answer]
you can map an array to 'loop'. It is declarative, kind of like React

**Reply by helalaou** [reply]
Thanks for the reply, Dolan!
Could you kindly share a snippet of how this can be done? I tried what you have mentioned (I found a similar discussion post) but It didn't work.

**Reply by helalaou** [reply]
Nevermind, I have solved it, thanks.

**Reply by dolanmiu** [reply]
Great!

**Comment by afzallakhani** [comment]
i need to create a table in that table number of rows need to be as per data in my array but when i try to map through it and create rows for each there is an error cannot read property cellCount of undefined.

is there any way to add exact number of rows to a table as per dynamic data from database?

**Reply by dolanmiu** [reply]
You can `array.map(d => ...)` as mentioned above
